### PR TITLE
database: introduce db_manager_test.go

### PR DESCRIPTION
**Pull request by ehnuje**
- To test DBManager interface for various implementations

## Proposed changes

- To increase test coverage of database package
- TODO commented tests will be covered by the next PR

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [x] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: storage/database/db_manager_test.go [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+
+		dbConfigs = append(dbConfigs, bc)
+		dbConfigs = append(dbConfigs, &badgerConfig)
+		dbConfigs = append(dbConfigs, &memoryConfig)
+	}
+
+	dbManagers = createDBManagers(dbConfigs)
+}
+
+// createDBManagers generate a list of DBManagers to test various combinations of DBConfig.
```
Comment (KimKyungup) [inline storage/database/db_manager_test.go]:
```suggestion
// createDBManagers generates a list of DBManagers to test various combinations of DBConfig.
```

File: storage/database/db_manager.go [hunk 2]
```diff
@@ -140,7 +139,7 @@ type DBManager interface {
 	WritePreimages(number uint64, preimages map[common.Hash][]byte)
 
 	// below operations are used in main chain side, not service chain side.
-	WriteChildChainTxHash(ccBlockHash common.Hash, ccTxHash common.Hash)
+	WriteChildChainTxHash(scBlockHash common.Hash, scTxHash common.Hash)
```
Comment (jimni1222) [inline storage/database/db_manager.go]: How about setting same name policy?

File: storage/database/db_manager_test.go [hunk 3]
```diff
@@ -0,0 +1,314 @@
+// Copyright 2018 The klaytn Authors
```
Comment (jimni1222) [inline storage/database/db_manager_test.go]:
```suggestion
// Copyright 2019 The klaytn Authors
```

File: storage/database/db_manager.go (hunk 2 above)
Comment (ehnuje) [reply storage/database/db_manager.go]: @jimni1222 Good point. @KimKyungup @jeonghonah How do you think about @jimni1222 suggest? ServiceChain or ChildChain?

File: storage/database/db_manager_test.go [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+
+// TestDBManager_CanonicalHash tests read, write and delete operations of canonical hash.
+func TestDBManager_CanonicalHash(t *testing.T) {
+	for _, dbm := range dbManagers {
+		// 1. Read from empty database, shouldn't be found.
+		assert.Equal(t, common.Hash{}, dbm.ReadCanonicalHash(0))
+		assert.Equal(t, common.Hash{}, dbm.ReadCanonicalHash(100))
+
+		// 2. Write a row to the database.
+
```
Comment (KimKyungup) [inline storage/database/db_manager_test.go]: unnecessary empty line.

File: storage/database/db_manager_test.go [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+		preimages2 := map[common.Hash][]byte{hash1: hash1[:], hash2: hash2[:]}
+		dbm.WritePreimages(num1, preimages2)
+
+		assert.Equal(t, hash1[:], dbm.ReadPreimage(hash1))
+		assert.Equal(t, hash2[:], dbm.ReadPreimage(hash2))
+	}
+}
+
+// TestDBManager_MainChain tests service chain related database operations, used in main chain.
```
Comment (KimKyungup) [inline storage/database/db_manager_test.go]:
```suggestion
// TestDBManager_ParentChain tests service chain related database operations, used in parent chain.
```
Please apply this modification. :)

File: storage/database/db_manager_test.go [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+
+		dbm.WriteLastIndexedBlockNumber(num1)
+		assert.Equal(t, num1, dbm.GetLastIndexedBlockNumber())
+
+		dbm.WriteLastIndexedBlockNumber(num2)
+		assert.Equal(t, num2, dbm.GetLastIndexedBlockNumber())
+	}
+}
+
+// TestDBManager_ServiceChain tests service chain related database operations, used in service chain.
```
Comment (KimKyungup) [inline storage/database/db_manager_test.go]:
```suggestion
// TestDBManager_ChildChain tests service chain related database operations, used in child chain.
```

File: storage/database/db_manager.go (hunk 2 above)
Comment (KimKyungup) [reply storage/database/db_manager.go]: ChildChain is right. :)

**Comment by ehnuje** [comment]
@jimni1222 Could you please review this change? I've reflected what you commented.

**Pull request merged**